binary-option/program/src/spl_utils.rs

```diff
@@ -0,0 +1,264 @@
+            token_program.key,
+            burn_account.key,
+            mint.key,
+            authority.key,
+            &[],
+            amount,
+        )?;
+        invoke(
+            &ix,
+            &[
+                burn_account.clone(),
+                mint.clone(),
+                authority.clone(),
+                token_program.clone(),
+            ],
+        )?;
+    }
+    Ok(())
+}
+
+pub fn spl_burn_signed<'a>(
+    token_program: &AccountInfo<'a>,
+    burn_account: &AccountInfo<'a>,
+    mint: &AccountInfo<'a>,
+    authority: &AccountInfo<'a>,
+    amount: u64,
+    signers: &[&[u8]],
+) -> ProgramResult {
+    msg!("Burn Signed");
+    if amount > 0 {
+        let ix = burn(
+            token_program.key,
+            burn_account.key,
+            mint.key,
+            authority.key,
+            &[&authority.key],
+            amount,
+        )?;
+        invoke_signed(
+            &ix,
+            &[
+                burn_account.clone(),
+                mint.clone(),
+                authority.clone(),
+                token_program.clone(),
+            ],
+            &[&signers],
+        )?;
+    }
+    Ok(())
+}
+
+pub fn spl_mint_to<'a>(
+    token_program: &AccountInfo<'a>,
+    dest_account: &AccountInfo<'a>,
+    mint: &AccountInfo<'a>,
+    authority: &AccountInfo<'a>,
+    amount: u64,
+    signers: &[&[u8]],
+) -> ProgramResult {
+    let ix = mint_to(
+        token_program.key,
+        mint.key,
+        dest_account.key,
+        authority.key,
+        &[&authority.key],
+        amount,
+    )?;
+    invoke_signed(
+        &ix,
+        &[
+            mint.clone(),
+            dest_account.clone(),
+            authority.clone(),
+            token_program.clone(),
+        ],
+        &[&signers],
+    )?;
+    Ok(())
+}
+
+pub fn spl_token_transfer<'a>(
+    token_program: &AccountInfo<'a>,
+    source: &AccountInfo<'a>,
+    destination: &AccountInfo<'a>,
+    owner: &AccountInfo<'a>,
+    amount: u64,
+) -> ProgramResult {
+    if amount > 0 {
+        let ix = transfer(
+            token_program.key,
+            source.key,
+            destination.key,
+            owner.key,
+            &[&owner.key],
+            amount,
+        )?;
+        invoke(
+            &ix,
+            &[
+                source.clone(),
+                destination.clone(),
+                owner.clone(),
+                token_program.clone(),
+            ],
+        )?;
+    }
+    Ok(())
+}
+
+pub fn spl_token_transfer_signed<'a>(
+    token_program: &AccountInfo<'a>,
+    source: &AccountInfo<'a>,
+    destination: &AccountInfo<'a>,
+    pda_account: &AccountInfo<'a>,
+    num: u64,
+    denom: u64,
+    signers: &[&[u8]],
+) -> ProgramResult {
+    if num > 0 && denom > 0 {
+        let ix = transfer(
+            token_program.key,
+            source.key,
+            destination.key,
+            pda_account.key,
+            &[&pda_account.key],
+            num / denom,
```

Comment:
This looks kind of strange / unsafe. In all cases but one, you pass `1` for `denom` anyway. Why not provide an `amount` to this function and handle the division safely where you actually need it?

Reply:
yeah I can make this change